photon-ml/src/main/scala/com/linkedin/photon/ml/model/Coefficients.scala

```diff
@@ -145,10 +147,13 @@ protected[ml] object Coefficients {
    */
   def apply(length: Int)(indices: Int*)(nnz: Double*): Coefficients = {
 
-    require(0 < length)
-    require(indices.length == nnz.length)
-    require(indices.sorted == indices)
-    // TODO: check for duplicates?
+    require(0 < length, s"Specified length < 0 for Coefficients: $length")
+    require(indices.length == nnz.length,
+      s"""There are ${indices.length} indices but ${nnz.length} non-zero specified for Coefficients.
+         | There should be as many indices as non-zero values.
+       """.stripMargin)
+    require(indices.sorted == indices, s"Indices need to be in strictly increasing order for Coefficients")
```

Comment:
For now, I'll remove the check from Coefficients.scala, pending a refactoring of the check in convertIndexAndValuePairArrayToSparseVector. Not having a preconditions in Coefficients directly is dangerous, as Coefficients should be a component that stands on its own, including the preconditions that verify its well-formedness. Separating the check is a recipe for having ill-formed Coefficients later upon refactoring.